### PR TITLE
Externalize identity provider config

### PR DESCRIPTION
**Pull request by LDMGN**
Replace compile-time configuration (environment.ts) with runtime configuration (external config.js available alongside index.html and other assets). Also replace dev configuration with staging in pipeline (pipeline is not yet parameterized to switch between staging and production).

### CODE REVIEW

**Comment by pschildkamp** [comment]
Don't we need an src/environments/environment.staging.ts where the production is set to true aswell? We have it for local and producion? Or do you use the *.prod.ts file for building the staging?

**Comment by LDMGN** [comment]
> Don't we need an src/environments/environment.staging.ts where the production is set to true aswell? We have it for local and producion? Or do you use the *.prod.ts file for building the staging?

The latter. Because `environment.ts` and `environment.production.ts` are not about the deployment environment, but about the build. So there is a development build and a production build (which can be different regarding debug logs, minification, etc).

Then there is deployment, which can be done to staging or production. And when deploying you generally always deploy a production build. So a production build (one using environment.production.ts) will go to both staging and (when given the green flag) to production.

File: src/app/services/littil-school/littil-school.service.spec.ts [hunk 1]
```diff
@@ -1,16 +1,11 @@
...
-  SpectatorHttp,
-} from '@ngneat/spectator';
-import { environment } from '../../../environments/environment';
+import { createHttpFactory, HttpMethod, SpectatorHttp, } from '@ngneat/spectator';
 import { School, SchoolService } from '../../api/generated';
 import { LittilSchoolService } from './littil-school.service';
 
 describe('LittilSchoolService', () => {
-  let baseUrl = environment.serverUrl;
+  let baseUrl = 'http://localhost:8080';
```
[the remark this answers is not in the document]
Comment (LDMGN) [reply src/app/services/littil-school/littil-school.service.spec.ts:8]: Hmm, good question. This particular test doesn't need the config, the URL should be arbitrary (right @Anjanas-dh ?). However, other tests might need it, so providing a `LittilConfigTestProvider` would be a good idea.

[the remark this answers is not in the document]
Comment (Anjanas-dh) [reply src/app/services/littil-school/littil-school.service.spec.ts:8]: Yeah it's a unit test, so it doesn't matter what you set here. You could also set: "a-api-url", it doesn't matter

File: src/app/services/littil-teacher/littil-teacher.service.spec.ts [hunk 2]
```diff
@@ -1,16 +1,11 @@
...
-  SpectatorHttp,
-} from '@ngneat/spectator';
-import { environment } from '../../../environments/environment';
+import { createHttpFactory, HttpMethod, SpectatorHttp, } from '@ngneat/spectator';
 import { GuestTeacherPostResource, TeacherService } from '../../api/generated';
 import { LittilTeacherService } from './littil-teacher.service';
 
 describe('LittilTeacherService', () => {
-  let baseUrl = environment.serverUrl;
+  let baseUrl = 'http://localhost:8080';
```
[the remark this answers is not in the document]
Comment (Anjanas-dh) [reply src/app/services/littil-teacher/littil-teacher.service.spec.ts:8]: See other answer

File: src/app/services/littil-user/littil-user.service.spec.ts [hunk 3]
```diff
@@ -1,16 +1,11 @@
...
-  SpectatorHttp,
-} from '@ngneat/spectator';
-import { environment } from '../../../environments/environment';
+import { createHttpFactory, HttpMethod, SpectatorHttp, } from '@ngneat/spectator';
 import { User, UsersService } from '../../api/generated';
 import { LittilUserService } from './littil-user.service';
 
 describe('LittilUserService', () => {
-  let baseUrl = environment.serverUrl;
+  let baseUrl = 'http://localhost:8080';
```
[the remark this answers is not in the document]
Comment (Anjanas-dh) [reply src/app/services/littil-user/littil-user.service.spec.ts:8]: See other answer

**Review by Anjanas-dh — APPROVED** [review]
Looking good!

**Pull request merged**